cruce.cpp

```diff
@@ -0,0 +1,100 @@
+#include<fstream>
+using namespace std;
+ifstream fin("cruce.in");
+ofstream fout("cruce.out");
+long long columnSum[501]={0};
+long long lineSum[501]={0};
+void read(int &n, int &k, int matrix[501][501])
+{
+    fin>>n>>k;
+    for(int index=1; index<=n; ++index)
+    {
+        for(int next=1; next<=n; ++next)
+        {
+            fin>>matrix[index][next];
+            lineSum[index]+=matrix[index][next];
+            columnSum[next]+=matrix[index][next];
+        }
+    }
+}
+
+
+
+void solveTask2(int k, int n, int matrix[501][501])
+{
+    int colIndex = 0, lineIndex = 0;
+    long long currentSum=0;
```

Comment:
`long long` este ok, dar in C++ modern preferam sa precizam explicit numarul de biti pentru fiecare tip de data. Aici vrei un tip de date pe 64 de biti, deci probabil ai putea sa scrii:
```
int64_t currentSum = 0;
```
Aici se vede clar ca vei avea 64 de biti indiferent pe ce platforma ruleaza programul.